src/ucp/core/ucp_ep.c

```diff
@@ -225,30 +225,42 @@ ucp_ep_local_disconnect_progress_remove_filter(const ucs_callbackq_elem_t *elem,
     return 1;
 }
 
-void ucp_ep_destroy_base(ucp_ep_h ep)
+static UCS_F_NOINLINE void ucp_ep_remove_progress_callbacks(ucp_ep_h ep)
 {
     ucp_worker_h worker = ep->worker;
-
-    if (--ep->ref_cnt != 0) {
-        return;
-    }
+    ucp_lane_index_t lane;
 
     /* Remove pending slow-path functions after all UCT EP lanes are destroyed,
      * because cleanup lanes purges all outstanding operation and purged
      * operations could add callbacks on the progress */
+    ucs_assert(ep->ref_cnt == 0);
+    if (ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL) {
+        for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+            ucs_assert((ep->uct_eps[lane] == NULL) ||
+                       ucp_is_uct_ep_failed(ep->uct_eps[lane]));
+        }
+    }
```

Comment:
i think this is redundant. we already check refcnt

Reply:
ok - done